### PR TITLE
fix(pam): Do not try to change the password twice but only after preliminary checks

### PR DESCRIPTION
**Pull request by 3v1n0**
As per pam_sm_chauthtok(3) PAM may call the function twice, with
different flags, so in such case just ignore the request and wait for
the next one to happen.

### CODE REVIEW

**Comment by 3v1n0** [comment]
Yeah, it's tested as part of #265, we could even add the same behavior to the cli now though...

**Review by denisonbarbosa — APPROVED** [review]

File: pam/pam.go [hunk 1]
```diff
@@ -193,7 +193,12 @@ func (h *pamModule) Authenticate(mTx pam.ModuleTransaction, flags pam.Flags, arg
 // ChangeAuthTok is the method that is invoked during pam_sm_chauthtok request.
 func (h *pamModule) ChangeAuthTok(mTx pam.ModuleTransaction, flags pam.Flags, args []string) error {
 	parsedArgs, logArgsIssues := parseArgs(args)
-	return h.handleAuthRequest(authd.SessionMode_PASSWD, mTx, flags, parsedArgs, logArgsIssues)
+	err := h.handleAuthRequest(authd.SessionMode_PASSWD, mTx, flags, parsedArgs, logArgsIssues)
+
```
Comment (didrocks) [inline pam/pam.go:197]: nitpick: remove empty blank line here

File: pam/pam.go [hunk 2]
```diff
@@ -210,6 +215,22 @@ func (h *pamModule) handleAuthRequest(mode authd.SessionMode, mTx pam.ModuleTran
 	}
 	logArgsIssues()
 
+	if mode == authd.SessionMode_PASSWD && flags&pam.PrelimCheck != 0 {
+		log.Debug(context.TODO(), "ChangeAuthTok, preliminary check")
+		_, closeConn, err := newClient(parsedArgs)
+		if err != nil {
+			log.Debugf(context.TODO(), "%s", err)
+			return fmt.Errorf("%w: %w", pam.ErrTryAgain, err)
```
Comment (didrocks) [inline pam/pam.go:223]: waow, double error wrapping, this is quite new! I wasn’t aware of this!

**Review by didrocks — APPROVED** [review]
If the tests covering this in the other PR (you linked the wrong one btw) is fine, then +1! Just look at my nitpick :)

Thanks for the double error wrapping, that will change my life from now on!

File: pam/pam.go (hunk 1 above)
Comment (3v1n0) [reply pam/pam.go:196]: This should be handled now, was lost in fixups...

**Comment by 3v1n0** [comment]
> If the tests covering this in the other PR (you linked the wrong one btw) is fine, then +1! Just look at my nitpick :)

Tests are included here too meanwhile, I've also rebased it.

> Thanks for the double error wrapping, that will change my life from now on!

Hehehe, at times can be problematic since errors may "overwrite" themselves when of the same type, but normally it's useful.

**Pull request merged**